src/xdp/xsk.c

```diff
@@ -4207,14 +4245,35 @@ XskGetOffloadWorker(
 
     TraceEnter(TRACE_XSK, "Xsk=%p", Xsk);
 
-    if (Xsk->Tx.Xdp.Queue == NULL) {
+    if (WorkItem->OffloadType == XdpTxOffloadChecksum && Xsk->Tx.Xdp.Queue == NULL) {
+        Status = STATUS_INVALID_DEVICE_STATE;
+        goto Exit;
+    }
+
+    if (WorkItem->OffloadType == XdpRxOffloadChecksum && Xsk->Rx.Xdp.Queue == NULL) {
+        Status = STATUS_INVALID_DEVICE_STATE;
+        goto Exit;
+    }
+
+    /*
+    TODO: on the RX path,
+
+    Do we want to move the RX interface offload handle creation deletion routines
+    to the XDP_RX_QUEUE creation / deletion paths (like how TX does it), or must we allocate / deallocate
+    the RX interface offload handle each time we attach/detach an interface?
+    In other words, can we get away with creating interface offload handle once in XDP_RX_QUEUE_CREATE
+    and delete it once in the deallocation path, or do we have to do it every time the app attaches/detaches
+    from an interface?
+    */
```

Comment:
just need an answer to this (ignore the other //!!!TODOs).

For now, I kept the logic for alloc/dealloc of the OffloadHandle in the attach/detach paths, and I have the below NULL check. 

This makes it so apps MUST attach a program first prior to querying the XSK_SOCKOPT_RX_OFFLOAD_CURRENT_CONFIG_CHECKSUM socket option.

Is this OK? @mtfriesen

Reply:
so to confirm, for all GET socket options, they are expected to succeed regardless of the state of the interface?

Reply:
Yup, that's the API layer expectation. Internally, we'll have to make XDP bind at least enough to the interface to return an answer.

Reply:
Addressed offline. The latest changes now respect the API contract.